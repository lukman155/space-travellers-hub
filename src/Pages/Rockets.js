import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  getRockets, toggleReserve,
} from '../Redux/rockets/rocketsSlice';

import './rocket.css';

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketList = useSelector((state) => state.rockets);

  useEffect(() => {
    if (rocketList.length) {
      return;
    }
    dispatch(getRockets());
  }, [dispatch, rocketList.length]);

  const content = rocketList.map((rocket) => (
    <article key={rocket.id}>
      <img className="rocket-img" src={rocket.flickr_images[0]} alt={rocket.name} />
      <div className="rocket-details">
        <h1>{rocket.rocket_name}</h1>
        <p>
          {rocket.reserved ? <span className="reserve-tag">Reserved</span> : ''}
          {rocket.description}
        </p>
        {!rocket.reserved
          ? <button onClick={() => dispatch(toggleReserve(rocket.id))} key={rocket.id} className="reserve-btn" type="button">Reserve Rocket</button>
          : <button onClick={() => dispatch(toggleReserve(rocket.id))} key={rocket.id} className="reserved" type="button">Cancel Reservation</button>}
      </div>
    </article>
  ));
  return (
    <section className="rockets-list">
      {content}
    </section>
  );
};
export default Rockets;
