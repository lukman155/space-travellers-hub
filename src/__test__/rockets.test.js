import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Rockets from '../Pages/Rockets';
import rocketsReducer from '../Redux/rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

describe('rockets', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Rockets />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
