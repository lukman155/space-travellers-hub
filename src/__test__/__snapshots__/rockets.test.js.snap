// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rockets renders correctly 1`] = `
<section
  className="rockets-list"
/>
`;
